discovery-provider/src/tasks/index_nethermind.py

```diff
@@ -333,10 +344,10 @@ def index_next_block(
         except NotAllTransactionsFetched as e:
             raise e
         try:
-            if next_block.number % 100 == 0:
+            if cast(int, next_block["number"]) % 100 == 0:
```

Comment:
wait but isn't this a `BlockNumber` which is an `int` under the hood anyway? is python type hinting this bad? 😢

Reply:
it _feels_ like this should be allowed without casts: https://docs.python.org/3/library/typing.html#newtype

Reply:
No, i'm just bad! it works!